modules/geolocation/geolocation.py

```diff
@@ -129,3 +129,40 @@ def calculate_pixel_to_geo_mapping(self):
 
         # Return matrix product of mappedGeoMatrix and sourcePixelMatrixInverse
         return (mappedGeoMatrix.dot(sourcePixelMatrixInverse))
+
+    def map_location_from_pixel(self, transformationMatrix, pixels):
+        """
+        Maps Geographical Location Coordinates in the destination image
+        
+        Parameters
+        -------
+            transformationMatrix : np.array(shape=(3,3))
+            pixels : np.array(shape=(5,2))
+
+        Returns
+        -------
+            np.array(shape=(5,2))
+        """
+
+        # Express all 2D coordinates of pixels as 3D coordinates with z value = 1
+        pixels = np.insert(pixels, 2, 1, axis = 1)
+
+        # Compute Homogeneous Coordinates: Product of Image Pixels and Coordinates
+        homogeneousCoordinates = np.matmul(transformationMatrix,pixels.T).T
+
+        geoCoordinates = np.empty(shape=(0, 2))
+        
+        # Cycle through all homogenized coordinates of pixels
+        for h in homogeneousCoordinates:
+            # Checking if the homogenized value of Z equals 0. If so, we return an empty array.
+            if h[2] == 0:
```

Comment:
I'm a bit worried about floating-point comparisons. Consider using something like `numpy.allclose()`. https://numpy.org/doc/stable/reference/generated/numpy.allclose.html

Reply:
Agreed. I have replaced the `np.testing.assert_almost_equal(actual, expected)` with `np.testing.assert_allclose(actual, expected, rtol = 1e-6)`. I set the relative tolerance of the assertion to 1e-6, which is a tolerance of about 0.0001%. All tests passed.

Reply:
No, what I meant was `if h[2] == 0:` on line 158. The unit tests were fine before.

Reply:
Sure thing. Changed `if h[2] == 0` to `np.allclose(h[2], 0)` and reverted tests back to `np.assert_almost_equal`.